isofit/radiative_transfer/radiative_transfer_engine.py

```diff
@@ -170,15 +176,31 @@ def __init__(
                 )
 
             # if necessary, resample prebuilt LUT to desired instrument spectral response
-            if not all(wl == self.lut.wl):
-                conv = xr.Dataset(coords={"wl": wl, "point": self.lut.point})
+            if not len(wl) == len(self.lut.wl) or all(wl == self.lut.wl):
+                conv = xr.Dataset(
+                    coords={"point": self.lut.point, "wl": wl},
+                    attrs={"RT_mode": self.rt_mode},
+                )
                 for quantity in self.lut:
-                    conv[quantity] = (
-                        ("wl", "point"),
-                        common.resample_spectrum(
-                            self.lut[quantity].data, self.lut.wl, wl, fwhm
-                        ),
-                    )
+                    if quantity in luts.Keys.alldim.keys():
+                        conv[quantity] = (
+                            ("point", "wl"),
+                            common.resample_spectrum(
+                                self.lut[quantity].data, self.lut.wl, wl, fwhm
+                            ),
+                        )
+                    if quantity == "solar_irr":
+                        conv[quantity] = (
+                            "wl",
+                            common.resample_spectrum(
+                                self.lut[quantity].data, self.lut.wl, wl, fwhm
+                            ),
+                        )
+                    if quantity == "fwhm":
+                        conv[quantity] = ("wl", fwhm)
+                    if quantity in luts.Keys.consts.keys():
+                        conv[quantity] = self.lut[quantity].data
```

Comment:
@jammont - can you look into utilize the WL dimension to make this choice (we'll need to exclude wl and fwhm keys).